### PR TITLE
Update README.md badges based on PR CI Labels

### LINKED ISSUE #1643: Add workflow Badges for the CI status of the develop branch
**Description**

As a global-workflow developer, I need to see the status of the current state of the CI builds and runs as GitHub badges in its **README.md**  display so I can easily see the state and stability of the develop branch.

**Requirements**
 
Use GitHub's Workflow Actions so we can generate the badges and specify their respective urls in the README.md file for the development branch

**Acceptance Criteria (Definition of Done)**

The badges work and correctly display the state that reflects the CI subsystem's generated labels:

Building $MACHINE - Running $MACHINE - Pass/Fail $MACINE

### PR DESCRIPTION
**Pull request by TerrenceMcGuinness-NOAA**
**Description**

Fixes #1643 

As a global-workflow developer, I need to see the status of the current state of the CI builds and runs as GitHub badges in its **README.md**  display so I can easily see the state and stability of the develop branch.

**Requirements**
 
Use GitHub's Workflow Actions so we can generate the badges in the README.md file for the development branch.
The GitHub Actions will update the badges only when a PR is merged to the **develop** branch and only update them occording to the state of the Labels in the PR itself.

This Action must be able to use the final state of the CI labels of the PR.   The labels must therefore be dynamically "pulled" from the PR just before it is merged.

Here is the map between the Label states and the Badges (for Orion and Hera respectively):

```
Pass -> Pass
Fail -> Fail
None -> Pending
```

For example, if the Badges are currently _Pass_ and a PR is merged without CI test Pass Label the corresponding Badge will be updated to _Pending_.

**Acceptance Criteria (Definition of Done)**

The badges work and correctly display the state of the develop branch in the **README.md** file that reflects the CI Labels of the PR just before it is merged.

- [x] Merged with test develop branch and Badges are displayed in test develop branch from the test merge
- [ ] Cycled test on Orion
- [ ] Forecast-only test on Hera
  
**Checklist**

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

### CODE REVIEW

**Review by aerorahul — APPROVED** [review]
looks good.

File: .github/workflows/hera.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+          OWNER: ${{ github.repository_owner }}
+          REPO_NAME: ${{ github.event.repository.name }}
+          PULL_REQUEST_NUMBER: ${{ github.event.pull_request.number }}
+        run: |
+          LABELS1="$(gh api repos/$OWNER/$REPO_NAME/pulls/$PULL_REQUEST_NUMBER --jq '.labels.[].name')"
+          LABELS=$(echo "$LABELS1" | tr '\n' ' ')
+          echo "labels=$LABELS" >> $GITHUB_OUTPUT
+          
+  passed:
+    if: contains( needs.getlabels.outputs.labels, 'CI-Hera-Passed') && github.event.pull_request.merged
```
Comment (WalterKolczynski-NOAA) [inline .github/workflows/hera.yaml:29]: Shouldn't the entire job just be enclosed within a `if: github.event.pull_request.merged`? There's no need to run any of it if the PR wasn't merged.

File: .github/workflows/hera.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    types: [closed]
+
+jobs:
+
+  getlabels:
+    runs-on: ubuntu-latest
+    outputs:
+      labels: ${{ steps.id.outputs.labels }}
+    steps:
+      - name: Get Lables Steps
```
Comment (WalterKolczynski-NOAA) [inline .github/workflows/hera.yaml]:
```suggestion
      - name: Get Label Steps
```

File: .github/workflows/orion.yaml
Comment (WalterKolczynski-NOAA) [inline .github/workflows/orion.yaml:1]: This is nearly identical to the Hera one. Can't we just have a single action that loops over the machines?

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]
Seems like there is no way to clear a pending status beyond merging another PR. A future PR should trigger CI testing on "workflow_dispatch" that will update the labels when complete.

File: .github/workflows/hera.yaml (hunk 1 above)
Comment (TerrenceMcGuinness-NOAA) [reply .github/workflows/hera.yaml:29]: Oh yes you're right, sort of, only that the `if:` syntax _is_ inside each job, but yes, after I pondered your question I found we could do this  (as you point out making all the `&&` logic for merging redundant):

```
on
  pull_request_review:
    types:
      - submitted
```

I'll have to test it though of course.

Comment (TerrenceMcGuinness-NOAA) [reply .github/workflows/hera.yaml:29]: Oh, point of order, ``passed:`` is a job.  The file has 4 jobs: one for getting the Labels dynamically, and the other three for getting the state of the badge for each case which are mutually exclusive.

Comment (TerrenceMcGuinness-NOAA) [reply .github/workflows/hera.yaml:29]: Oh but darn, I'm not sure there is a `pull_request_review_target` and we need the `target` for this to work on forked PRs.

Comment (TerrenceMcGuinness-NOAA) [reply .github/workflows/hera.yaml:29]: Ok, took a deeper dive, here is a GitHub discussion that came to the same conclusion in my original logic that have with this: ([discussion](https://github.com/orgs/community/discussions/26724)).

File: .github/workflows/orion.yaml
Comment (TerrenceMcGuinness-NOAA) [reply .github/workflows/orion.yaml:1]: Note that I can see - not sure how we could "loop" over the `with:` structure value pairs with variables.  These are YAML files BTW not code albeit the run section has code in it.

**Review by TerrenceMcGuinness-NOAA — COMMENTED** [review]
bunch of comments with answers

File: .github/workflows/orion.yaml
Comment (WalterKolczynski-NOAA) [reply .github/workflows/orion.yaml:1]: https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

**Review by TerrenceMcGuinness-NOAA — COMMENTED** [review]
suggestions are nice but not actionable

File: .github/workflows/orion.yaml
Comment (TerrenceMcGuinness-NOAA) [reply .github/workflows/orion.yaml:1]: Oh there it is!  Yes, I had saw "matrix" and had been glossing over it.  Looks like correct solution.  See that's why we have reviews!!

Comment (WalterKolczynski-NOAA) [reply .github/workflows/orion.yaml:1]: Deferred to a future PR.

**Review by WalterKolczynski-NOAA — APPROVED** [review]

**Pull request merged**